solutions/devsprint-firas-safa-1/GitHubApp/Data/Network/NetworkService.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  NetworkEngine.swift
+//  GitHubApp
+//
+//  Created by Razee Hussein-Jamal on 09/02/2022.
+//
+
+import Foundation
+
+
+protocol NetworkRequestProtocol {
+    func makeGetRequest<ResponseType: Decodable>(with request: URLRequest, using session: URLSession, completion: @escaping (GithubResult<ResponseType>) -> Void)
+}
+
+class NetworkEngine: NetworkRequestProtocol {
```

Comment:
- create a private variable called session
- use the initializer function to inject the session from outside (you can use a default value of shared)

Reply:
Always use a struct